2008-05-11  Aurelien Jarno  <aurelien@aurel32.net

	* sysdeps/unix/sysv/linux/alpha/sys/procfs.h (ELF_NGREG,
	ELF_NFPREG, elf_greg_t, elf_gregset_t, elf_fpreg_t,
	elf_fpregset_t): Define. Don't include asm/elf.h.

---
 ports/sysdeps/unix/sysv/linux/alpha/sys/procfs.h |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

--- a/ports/sysdeps/unix/sysv/linux/alpha/sys/procfs.h
+++ b/ports/sysdeps/unix/sysv/linux/alpha/sys/procfs.h
@@ -29,10 +29,23 @@
 #include <sys/types.h>
 #include <sys/ucontext.h>
 #include <sys/user.h>
-#include <asm/elf.h>
 
 __BEGIN_DECLS
 
+/*
+ * The OSF/1 version of <sys/procfs.h> makes gregset_t 46 entries long.
+ * I have no idea why that is so.  For now, we just leave it at 33
+ * (32 general regs + processor status word).
+ */
+#define ELF_NGREG       33
+#define ELF_NFPREG      32
+
+typedef unsigned long elf_greg_t;
+typedef elf_greg_t elf_gregset_t[ELF_NGREG];
+
+typedef double elf_fpreg_t;
+typedef elf_fpreg_t elf_fpregset_t[ELF_NFPREG];
+
 struct elf_siginfo
   {
     int si_signo;			/* Signal number.  */
